### PR TITLE
Remove package name validation from classpath scanning

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes: https://github.com/junit-team/junit5/issues/2570

### CODE REVIEW

**Review by Cu2y — APPROVED** [review]

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java [hunk 1]
```diff
@@ -149,11 +149,16 @@ public static void assertPackageNameIsValid(String packageName) {
 	 */
 	static class JavaNameUtils {
 
+		/*
+		 * Based on https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-Identifier
+		 * */
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java:154]:
```suggestion
		/*
		 * Based on https://docs.oracle.com/javase/specs/jls/se16/html/jls-3.html#jls-Identifier
		 */
```

**Review by marcphilipp — CHANGES_REQUESTED** [review]

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java [hunk 2]
```diff
@@ -70,7 +71,6 @@
 
 	List<Class<?>> scanForClassesInPackage(String basePackageName, ClassFilter classFilter) {
 
-		PackageUtils.assertPackageNameIsValid(basePackageName);
 		Preconditions.notNull(classFilter, "classFilter must not be null");
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClasspathScanner.java:74]:
```suggestion
		Preconditions.condition(DEFAULT_PACKAGE_NAME.equals(basePackageName) || isNotBlank(basePackageName), "basePackageName must not be blank");
		Preconditions.notNull(classFilter, "classFilter must not be null");
```

**Review by marcphilipp — CHANGES_REQUESTED** [review]
Sorry, stating that `PackageUtils` should be removed was premature. What I meant was to remove `PackageUtils.assertPackageNameIsValid`.

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
Thanks, @paplorinc! 👍

**Pull request merged**